multi-valued index in mysql 8.x can't be synced to downstream tidb cluster by DM

### What did you do?

in upstream mysql 8.0
create table include jason cloumn
add multi-valued index on jason key-value
```sql
mysql> show create table mvindex.user_data1;
+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table      | Create Table                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| user_data1 | CREATE TABLE `user_data1` (
  `uid` bigint NOT NULL AUTO_INCREMENT,
  `j` json DEFAULT NULL,
  `j2` json DEFAULT NULL,
  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `update_time` datetime DEFAULT NULL,
  PRIMARY KEY (`uid`),
  UNIQUE KEY `j2` ((cast(json_extract(`j2`,_utf8mb4'$[*]') as signed array)),`uid`),
  KEY `k_1` (`update_time`)
) ENGINE=InnoDB AUTO_INCREMENT=1000001 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
+------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```
in downstream tidb
check data and schema can be synced

### What did you expect to see?

data and schema can be synced

### What did you see instead?

```sql
           "subTaskStatus": [
                {
                    "name": "dm_with_mvindex",
                    "stage": "Paused",
                    "unit": "Sync",
                    "result": {
                        "isCanceled": false,
                        "errors": [
                            {
                                "ErrCode": 36067,
                                "ErrClass": "sync-unit",
                                "ErrScope": "internal",
                                "ErrLevel": "high",
                                "Message": "startLocation: [position: (binlog.000004, 355718121), gtid-set: ebf9c552-dd9a-11ed-8a0e-eea94
753c9ce:1-663], endLocation: [position: (binlog.000004, 355718306), gtid-set: ebf9c552-dd9a-11ed-8a0e-eea94753c9ce:1-664], origin SQL: [a
lter table mvindex.user_data1 add unique index j2((cast(j2-\u003e'$[*]' as signed array)),uid)]: parse DDL: alter table mvindex.user_data
1 add unique index j2((cast(j2-\u003e'$[*]' as signed array)),uid)",
                                "RawCause": "line 1 column 83 near \"array)),uid)\" ",
                                "Workaround": "Please confirm your DDL statement is correct and needed. For TiDB compatible DDL, see http
s://docs.pingcap.com/tidb/stable/mysql-compatibility#ddl. You can use `handle-error` command to skip or replace the DDL or add a binlog f
ilter rule to ignore it if the DDL is not needed."
                            }
                        ],
                        "detail": null
                    },
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

change to master version:
dm paniced ;
``` sql
starting dm-worker ...
/dm-worker --name=dm-dm-worker-0 --join=dm-dm-master:8261 --advertise-addr=dm-dm-worker-0.dm-dm-worker-peer:8262 --worker-addr=0.0.0.0:8262 --config=/etc/dm-worker/dm-worker.toml

[2023/04/18 08:40:43] [info] binlogsyncer.go:173 create BinlogSyncer with config {429547710 mysql upstream1 3306 root  dm-dm-worker-0  false false <nil> false +00:00 false 0 30s 1m0s 1 true true 0 <nil> 0xc001436f60 0x792520 0x30f08a0 false}
[2023/04/18 08:40:43] [info] binlogsyncer.go:428 begin to sync binlog from GTID set 7c17f7ee-ddbe-11ed-8af5-3249481d04ac:1-655
[2023/04/18 08:40:43] [info] binlogsyncer.go:813 rotate to (binlog.000004, 4)
panic: runtime error: index out of range [5] with length 5

goroutine 447 [running]:
github.com/pingcap/tiflow/pkg/sqlmodel.getColsAndValuesOfIdx(...)
        github.com/pingcap/tiflow/pkg/sqlmodel/utils.go:32
github.com/pingcap/tiflow/pkg/sqlmodel.(*RowChange).getCausalityString(0xc001854980, {0xc000d04690?, 0x5, 0x5})
        github.com/pingcap/tiflow/pkg/sqlmodel/causality.go:147 +0xa70
github.com/pingcap/tiflow/pkg/sqlmodel.(*RowChange).CausalityKeys(0xc001854980)
        github.com/pingcap/tiflow/pkg/sqlmodel/causality.go:36 +0xb2
github.com/pingcap/tiflow/dm/syncer.(*causality).run(0xc0006331a0)
        github.com/pingcap/tiflow/dm/syncer/causality.go:83 +0x2b3
github.com/pingcap/tiflow/dm/syncer.causalityWrap.func1()
        github.com/pingcap/tiflow/dm/syncer/causality.go:60 +0x25
created by github.com/pingcap/tiflow/dm/syncer.causalityWrap
        github.com/pingcap/tiflow/dm/syncer/causality.go:59 +0x3a5
lt@ltdebijndiannao test-plan % 
```

--- a/pkg/sqlmodel/causality.go
+++ b/pkg/sqlmodel/causality.go
@@ -144,6 +144,11 @@ func (r *RowChange) getCausalityString(values []interface{}) []string {
 	ret := make([]string, 0, len(pkAndUks))
 
 	for _, indexCols := range pkAndUks {
+		// TODO: should not support multi value index and generate the value
+		// TODO: also fix https://github.com/pingcap/tiflow/issues/3286#issuecomment-971264282
+		if indexCols.MVIndex {
+			continue
+		}
 		cols, vals := getColsAndValuesOfIdx(r.sourceTableInfo.Columns, indexCols, values)
 		// handle prefix index
 		truncVals := truncateIndexValues(r.tiSessionCtx, r.sourceTableInfo, indexCols, cols, vals)

--- a/pkg/sqlmodel/row_change_test.go
+++ b/pkg/sqlmodel/row_change_test.go
@@ -275,7 +275,7 @@ func (s *dpanicSuite) TestExpressionIndex() {
 	sql := `CREATE TABLE tb1 (
     	id INT PRIMARY KEY,
     	j JSON,
-    	KEY j_index ((cast(json_extract(j,'$[*]') as signed array)), id)
+    	UNIQUE KEY j_index ((cast(json_extract(j,'$[*]') as signed array)), id)
 )`
 	ti := mockTableInfo(s.T(), sql)
 	change := NewRowChange(source, nil, nil, []interface{}{1, `[1,2,3]`}, ti, nil, nil)
@@ -283,6 +283,10 @@ func (s *dpanicSuite) TestExpressionIndex() {
 	s.Equal("INSERT INTO `db`.`tb1` (`id`,`j`) VALUES (?,?)", sql)
 	s.Equal([]interface{}{1, `[1,2,3]`}, args)
 	require.Equal(s.T(), 2, change.ColumnCount())
+	keys := change.CausalityKeys()
+	// TODO: need change it after future fix
+	require.Equal(s.T(), []string{"1.id.db.tb1"}, keys)
+
 	change2 := NewRowChange(source, nil, []interface{}{1, `[1,2,3]`}, []interface{}{1, `[1,2,3,4]`}, ti, nil, nil)
 	sql, args = change2.GenSQL(DMLUpdate)
 	s.Equal("UPDATE `db`.`tb1` SET `id` = ?, `j` = ? WHERE `id` = ? LIMIT 1", sql)

